### PR TITLE
[chore] Cancellation token best practice, parameter cleanup

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

- Add optional cancellation token parameter to all HTTP functions, to allow end users to cancel operations if needed. This is [standard practice for async functions](https://medium.com/geekculture/how-to-support-the-http-request-cancelation-in-net-core-4b5561d9605d), especially network-related ones.
  - Stripe does it: https://github.com/stripe/stripe-dotnet/blob/c8bc4fce820ed394de85254a7e5a8302fe7b8182/src/Stripe.net/Services/TestHelpers/Issuing/Cards/CardService.cs#L28
- For functions that use the Parameters objects rather than dictionaries, consolidate all explicit parameters into the Parameters object, so that all functions accepting Parameters objects have one of these two signatures:
```csharp
Create(myParameterSet)
```

```csharp
Update(objectId, myParameterSet)
```
  - This only impacts the `All` function for retrieving all reports, which had an explicit separate `type` parameter that is now part of the `Report.All` parameter set passed into the function.
- Parameter-object functions no longer redirect to the Dictionary equivalent, but instead are self-contained. This makes them easier to maintain and read, and will make migration easier when Dictionary functions are eventually dropped entirely.
- Removed default "pdf" value for file format for labels and scan forms. Users must now explicitly declare what format they want.
- Removed ability to use `Retrieve` function in User service without passing in an ID to retrieve self.
  - `Retrieve` now requires a user ID be provided. To retrieve self, users should use the `RetreiveMe` function explicitly.
 

# Testing

- Updated unit tests as needed (file format required parameter)
  - No cassettes needed re-recording, suggesting that parameter changes do not impact the final HTTP request

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: EasyPost.Tests/BetaFeaturesTests/ServicesTests/ReportServiceTest.cs [hunk 1]
```diff
@@ -85,11 +88,12 @@ public async Task TestCreateWithColumns()
                 { "columns", columns },
                 { "start_date", Fixtures.ReportDate },
                 { "end_date", Fixtures.ReportDate },
+                { "type", "shipment" },
```
Comment (Justintime50) [inline EasyPost.Tests/BetaFeaturesTests/ServicesTests/ReportServiceTest.cs:91]: question: why make this change? Does it better align with the other libs?

File: EasyPost/Services/UserService.cs [hunk 2]
```diff
@@ -40,26 +41,26 @@ public async Task<User> CreateChild(Dictionary<string, object> parameters)
...
-        public async Task<User> CreateChild(BetaFeatures.Parameters.Users.CreateChild parameters)
+        public async Task<User> CreateChild(BetaFeatures.Parameters.Users.CreateChild parameters, CancellationToken cancellationToken = default)
         {
             // Because the normal CreateChild method does wrapping internally, we can't simply pass the parameters object to it, otherwise it will wrap the parameters twice.
-            return await RequestAsync<User>(Method.Post, "users", parameters.ToDictionary());
+            return await RequestAsync<User>(Method.Post, "users", cancellationToken, parameters.ToDictionary());
         }
 
         /// <summary>
-        ///     Retrieve a User from its id. If no id is specified, it returns the user for the api_key specified.
```
Comment (Justintime50) [inline EasyPost/Services/UserService.cs:50]: issue: I don't think we want to remove this functionality. Users have come to rely on it and our docs clearly state this is a reasonable thing to do.

**Review by Justintime50 — COMMENTED** [review]
Let's not optimize too much more. We are getting back into the weeds here a bit.

**Review by jchen293 — APPROVED** [review]

File: EasyPost.Tests/BetaFeaturesTests/ServicesTests/ReportServiceTest.cs (hunk 1 above)
Comment (nwithan8) [reply EasyPost.Tests/BetaFeaturesTests/ServicesTests/ReportServiceTest.cs:91]: Helps make the Parameters set consistent. This was the only function that takes an explicit parameter + a Parameters object.

File: EasyPost/Services/UserService.cs (hunk 2 above)
Comment (nwithan8) [reply EasyPost/Services/UserService.cs:50]: Fair enough, reverting.

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**